### PR TITLE
Community

### PR DESCRIPTION
**Pull request by AibekYrysbekov**
This code allows you to extract pull request information from a GitHub repository using the GitHub API and store this data in a SQLite database for later use or analysis.

### CODE REVIEW

File: community/pmc-tool/blog/views.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    if not pull_requests:
+        break
+
+    # Extract list of pull request authors
+    authors = [pr['user']['login'] for pr in pull_requests]
+
+    for author in authors:
+        if author not in authors_count:
+            authors_count[author] = 0
+        authors_count[author] += 1
```
Comment (pabloem) [inline community/pmc-tool/blog/views.py:44]: let's put this into a function that returns a list of pull requests. Like so:

```
def get_prs(owner,  repo, open_only=True, token) -> list[dict]:
  # Get the list of PRs and return all the PRs
```

File: community/pmc-tool/blog/views.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+c.execute('SELECT author, count FROM pull_requests')
+results = c.fetchall()
+
+for row in results:
+    author, count = row
+    print(f'Author: {author}, PRs: {count}')
+# Commit the changes and close the database connection
+conn.commit()
+conn.close()
```
Comment (pabloem) [inline community/pmc-tool/blog/views.py:58]: Let's move everything related to the database connection to a new file. Maybe we can call it `db.py`. We can have a few functions:

```
def create_db_connection(...):
   ...

def create_pull_requests_table(conn):
    ...

def insert_prs_to_db(conn, prs: list[dict]):
  ....
```

File: community/pmc-tool/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+render==1.0.0
```
Comment (pabloem) [inline community/pmc-tool/requirements.txt:1]: we don't need `render`, right? we do need requests:

```suggestion
requests
```

**Review by pabloem — COMMENTED** [review]
Thanks! This looks much better. Let's rename the folder `pmc-tool/blog/` into `pmc-tool/collector` - and we will keep the code for data collection in that folder (i.e. downloading data from github).

File: community/pmc-tool/blog/views.py (hunk 1 above)
Comment (AibekYrysbekov) [reply community/pmc-tool/blog/views.py]: I don't understand what I need to do here

File: community/pmc-tool/collector/views.py [hunk 4]
```diff
@@ -51,25 +50,20 @@
...
+create_pull_requests_table(conn)
 
 # Retrieve list of pull requests
-authors_count = dict()
+authors_count = {}
+
 
 for page_number in range(1, 8):
-    response = requests.get(api_endpoint + f"?page={page_number}", headers=headers, params={'state': 'open'})
+    response = requests.get(API_ENDPOINT + f"?page={page_number}", headers=headers, params={'state': 'open'})
```
Comment (pabloem) [inline community/pmc-tool/collector/views.py]: please create a new PR and put this logic in a new function called `query_prs`

File: community/pmc-tool/collector/main.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    if not pull_requests:
+        break
+
+    # Extract list of pull request authors
+    authors = [pr['user']['login'] for pr in pull_requests]
+
+    for author in authors:
+        if author not in authors_count:
+            authors_count[author] = 0
+        authors_count[author] += 1
```
Comment (pabloem) [inline community/pmc-tool/collector/main.py:78]: please move all of this logic to a function called `query_prs`. Probably in a new file called... `gh.py` - in that file we will put all of the code that downloads from github.

**Pull request merged**